src/layout-rect.js

```diff
@@ -109,8 +109,8 @@ export function layoutRectFromDomRect(rect) {
 
 /**
  * Returns true if the specified two rects overlap by a single pixel.
- * @param {!LayoutRectDef} r1
- * @param {!LayoutRectDef} r2
+ * @param {{top: number, bottom: number, left: number, right: number}} r1
+ * @param {{top: number, bottom: number, left: number, right: number}} r2
```

Comment:
This allows being called with ClientRect since `x` and `y` are not needed anyways.

Reply:
No obfuscation problems in single pass?

Reply:
Fixed.